ext/imap/php_imap.c

```diff
@@ -3940,22 +4281,83 @@ static void _php_make_header_object(zval *myzvalue, ENVELOPE *en)
 
 	object_init(myzvalue);
 
-	if (en->remail) add_property_string(myzvalue, "remail", en->remail);
-	if (en->date) add_property_string(myzvalue, "date", (char*)en->date);
-	if (en->date) add_property_string(myzvalue, "Date", (char*)en->date);
-	if (en->subject) add_property_string(myzvalue, "subject", en->subject);
-	if (en->subject) add_property_string(myzvalue, "Subject", en->subject);
-	if (en->in_reply_to) add_property_string(myzvalue, "in_reply_to", en->in_reply_to);
-	if (en->message_id) add_property_string(myzvalue, "message_id", en->message_id);
-	if (en->newsgroups) add_property_string(myzvalue, "newsgroups", en->newsgroups);
-	if (en->followup_to) add_property_string(myzvalue, "followup_to", en->followup_to);
-	if (en->references) add_property_string(myzvalue, "references", en->references);
+	if (en->remail) {
+		zend_update_property_string(
+			Z_OBJCE_P(myzvalue), Z_OBJ_P(myzvalue),
+			"remail", strlen("remail"),
+			en->remail
+		);
+	}
+	if (en->date) {
+		zend_update_property_string(
+			Z_OBJCE_P(myzvalue), Z_OBJ_P(myzvalue),
+			"date", strlen("date"),
+			(char*)en->date
+		);
+		zend_update_property_string(
+			Z_OBJCE_P(myzvalue), Z_OBJ_P(myzvalue),
+			"Date", strlen("Date"),
```

Comment:
I don't really know the reason either, probably because the casing was already inconsistent. But this just makes it weirder as sometimes it has the first character uppercased, otherwise no.

Ideally we would have named classes with fixed properties. But I can't really work on ext/imap any more as Fedora has dropped c-client from the libs due to how old and unmaintained it is.

Maybe we should just drop ext/imap....